src/utils/storage-helper.js

```diff
@@ -1,13 +1,21 @@
 import { IMAGE_DOWN_SCALING_WIDTH } from '../domain/constants.js';
 
-const getScaledDimensions = (img, width) => {
+const getScaledDownDimensions = (img, width) => {
+  if (img.naturalWidth <= width) {
+    return { width: img.naturalWidth, height: img.naturalHeight };
+  }
   const ratio = img.naturalWidth / width;
   return { width, height: Math.round(((img.naturalHeight / ratio) + Number.EPSILON) * 100) / 100 };
 };
 
 const scalableFileTypes = ['image/png', 'image/jpg', 'image/jpeg', 'image/webp'];
 
-const scaleDownImage = ({ file, width }) => {
+const shouldOptimizeImage = ({ naturalSize, naturalWidth, width }) => {
+  const optimize = (naturalWidth > width && (naturalSize > 500 * 1000)) || (naturalSize > 1.5 * 1000 * 1000);
+  return optimize;
+};
+
+const optimizeImage = ({ file, width }) => {
```

Comment:
Just a small thing: I think it might better easier to understand immediately, if this param would be called something like e.g. `widthThreshold` or `optimalWidth`, what do you think?

Reply:
Ended up using the constant directly and no longer passing it as an argument.